model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -182,6 +182,48 @@ message ProcessBundleDescriptor {
   org.apache.beam.model.pipeline.v1.ApiServiceDescriptor state_api_service_descriptor = 7;
 }
 
+// Represents a partition of the bundle into two bundles: a "primary" and
+// a "residual", with the following properties:
+// - The work in primary and residual doesn't overlap, and combined, adds up
+//   to the work in the current bundle if the split hadn't happened.
+// - The current bundle, if it keeps executing, will have done none of the
+//   work under residual roots.
+// - The current bundle, if no further splits happen, will have done exactly
+//   the work under primary_roots.
+// For more rigorous definitions see https://s.apache.org/beam-breaking-fusion
+message BundleSplit {
+  // One of the root applications specifying the scope of work for a bundle.
+  message Application {
+    // (Required) The primitive transform to which to pass the element
+    string ptransform_id = 1;
+
+    // (Required) Name of the transform's input to which to pass the element.
+    string input_id = 2;
+
+    // (Required) The encoded element to pass to the transform.
+    bytes element = 3;
+
+    // Approximate lower bounds on timestamps of elements that this PTransform
+    // will produce into each of its output PCollections, when invoked on this
+    // element. Keyed by the transform's local output name.
+    map<string, int64> output_watermarks = 4;
+
+    // Approximate fraction of all work of the current bundle (before split)
+    // represented by invoking this Application and its downstream applications.
+    // The sum of fraction_of_work between all primary_roots and residual_roots
+    // must add up to approximately 1.0.
+    DoubleValue fraction_of_work = 5;
```

Comment:
Why DoubleValue and not double?

I think we should always ask for this from the SDK.

Reply:
I decided to use DoubleValue because there is a difference between specifying 0 and specifying "I don't know" - if an SDK fails to specify this, I'd like to be able to tell whether there's a 0 because the SDK doesn't now, or whether it's because the SDK really thinks it's 0.

Reply:
That would mean that this value is `optional`. If it becomes a required field we should change it back to double.